### ISSUE TITLE
TypeError: serviceBlock is undefined

### ISSUE THREAD

**Issue by OvalMedia** [netresearch/dhl-module-shipping-m2#60]
**Bug description**
When entering the checkout I get a JS error: "TypeError: serviceBlock is undefined" referencing line 41 in validate-service-compatibility.js

serviceBlock is supposed to be "serviceBlock = registry.get({component: 'Dhl_Shipping/js/view/checkout/shipping/service-block'});"

I checked, Dhl_Shipping/js/view/checkout/shipping/service-block file DOES exist.

It is NOT possible to advance to the next step.

**Steps to reproduce**
To reproduce the problem:
1. Install module
2. Got to checkout
3. See error message ... in the js console

**Expected behavior**
Add the expected behavior. In other words: what SHOULD happen, but doesn't.

**Screenshots / logs**
If possible, add screenshots or logs to clarify your problem.

**Environment**
 - Magento version: 2.2.8
 - PHP version: 7.1
 - Module version:  0.10.2 (composer), 0.10.0 (module.xml)

**FAQ**
Has the [Knowledge Base](http://dhl.support.netresearch.de) been checked for a solution?
- [x] Yes, but couldn't find anything
- [ ] Yes, but solution did not work

**Comment by ngolatka** [comment]
@OvalMedia Are you using some kind of third-party-checkout, like OneStepCheckout or anything similar?

**Comment by OvalMedia** [comment]
No, I don't (yet).

**Comment by ngolatka** [comment]
I don't have a running M2.2.8 with DHL Shipping 0.10.2 (or 0.10.0) available right now.

But I do have an M2.3.1 with DHL Shipping 0.10.3, and everything works totally fine there. I don't think the Mage version does make a difference here, and the module changelog does not say anything about an issue in this area.

Could you please [file a support request](http://dhl.support.netresearch.de/support/tickets/new)? I think this is probably not a problem in the DHL module, but something else.

**Comment by OvalMedia** [comment]
You are right. I just set up a fresh 2.2.9 and there are no issues. In the meantime I upgraded the shop in question to 2.2.9 as well but the problem persists. I will try to isolate the issue and get in touch with your support. Thank you.

**Comment by OvalMedia** [comment]
Turned out to be an incompatibility with [Mageplaza's One Step Checkout module](https://www.mageplaza.com/magento-2-one-step-checkout-extension/).

In the file

`view/frontend/web/js/view/checkout/shipping/service-block.js`

change this...

```
var countryId = quote.shippingAddress().countryId;
var carrierCode = quote.shippingMethod().carrier_code + '_' + quote.shippingMethod().method_code;
var postalCode = quote.shippingAddress().postcode;
```

... to this ...

```
var countryId = quote.shippingAddress() ? quote.shippingAddress().countryId : null;
var carrierCode = quote.shippingMethod() ? (quote.shippingMethod().carrier_code + '_' + quote.shippingMethod().method_code) : null;
var postalCode = quote.shippingAddress() ? quote.shippingAddress().postcode : null;
```

**Comment by powli** [comment]
Hey @OvalMedia, if you have the fix already, feel free to submit a pull request :slightly_smiling_face:

**Comment by ngolatka** [comment]
Thanks for the detailed explanation @OvalMedia I've made some small clarifications to your [above comment](60#issuecomment-511716837) to help future users.

**Comment by kolaente** [comment]
We're having the same issue as  @OvalMedia, the mageplaza support gave us the exact same fix. Will there be a patch release to fix this?

**Comment by kolaente** [comment]
I've put up a pr: #66

**Comment by powli** [comment]
Included in [0.11.0](https://github.com/netresearch/dhl-module-shipping-m2/releases/tag/0.11.0)
